src/analyze/gui/parser.cpp

```diff
@@ -647,16 +658,37 @@ void Parser::parseImpl(const QString& path, const QString& diffBase, const Filte
     auto oldData = std::move(m_data);
     using namespace ThreadWeaver;
     stream() << make_job([this, oldData, path, diffBase, filterParameters]() {
+        const auto isReparsing = (path == m_path && oldData && diffBase.isEmpty());
+        auto parsingMsg = isReparsing ? i18n("reparsing data") : i18n("parsing data");
+        emit progressMessageAvailable(parsingMsg);
+
+        auto updateProgress = [this, parsingMsg](const ParserData & data){
+            const auto numPasses = data.stringCache.diffMode ? 2 : 3;
+            auto passCompletion = 1.0 * data.parsingState.readCompressedByte / data.parsingState.fileSize;
+            auto totalCompletion = (data.parsingState.pass + passCompletion)/numPasses;
+            auto spentTime_s = data.parseTimer.elapsed() / 1000.0; // elapsed is in ms
+            auto totalRemainingTime_s = (spentTime_s / totalCompletion) * (1.0 - totalCompletion);
+            auto message = QString(
```

Comment:
this is still a string puzzle that's hard to localize. better use something like this here:

```
auto message = i18n("%1, pass: %2/%3, elapsed: %4, remaining: %5",
    parsingMsg, data.parsingState.pass + 1, numPasses,
    Util::formatTime(spentTime),
    Util::formatTime(totalRemainingTime));
```